app/javascript/dashboard/components-next/captain/pageComponents/document/CreateDocumentDialog.vue

```diff
@@ -2,28 +2,43 @@
 import { ref } from 'vue';
 import { useStore } from 'dashboard/composables/store';
 import { useAlert } from 'dashboard/composables';
-import { useI18n } from 'vue-i18n';
-
 import Dialog from 'dashboard/components-next/dialog/Dialog.vue';
 import DocumentForm from './DocumentForm.vue';
 
-const emit = defineEmits(['close']);
-const { t } = useI18n();
+const emit = defineEmits(['close', 'success']);
 const store = useStore();
 
 const dialogRef = ref(null);
 const documentForm = ref(null);
 
-const i18nKey = 'CAPTAIN.DOCUMENTS.CREATE';
-
 const handleSubmit = async newDocument => {
```

Comment:
Let's split this function, the formData creation can go outside the component too, we can let `document.js` handle it